compiler/rustc_infer/src/infer/relate/nll.rs

```diff
@@ -557,11 +557,7 @@ where
         match b.kind() {
             ty::ConstKind::Infer(InferConst::Var(_)) if D::forbid_inference_vars() => {
                 // Forbid inference variables in the RHS.
-                self.infcx.dcx().span_delayed_bug(
-                    self.delegate.span(),
-                    format!("unexpected inference var {b:?}",),
-                );
-                Ok(a)
+                self.infcx.dcx().bug(format!("unexpected inference var {b:?}"));
```

Comment:
`bug` vs `span_bug`?

Reply:
You want me to use `span_bug`? What span would I use?

Reply:
`self.delegate.span()`? the same we previously used in the `span_delayed_bug` :thinking:

Reply:
Oh, I removed that because of [this comment](https://github.com/rust-lang/rust/pull/121208#discussion_r1494204882). Maybe I misinterpreted what you were asking for.

Reply:
ah xd, yeah :grin: i messed that up :sweat_smile: didn't look at the context in my second review :sweat_smile: sorry